tempesta_fw/pool.h

```diff
@@ -28,9 +28,14 @@
 #define TFW_POOL_ZERO	0x1
 
 typedef struct {
-	unsigned char *base; /* base address of the pool */
-	unsigned int order; /* total size of the pool */
+	struct list_head list;
```

Comment:
Do we really need double linked list? Probably we just should keep head and tail of the list... Double linked lists are too complex for fast synchronization (e.g. lock-free techniques).